test/jdk/java/lang/invoke/indify/Indify.java

```diff
@@ -652,20 +706,339 @@ private void removeEmptyJVMSlots(List<Object> args) {
             for (;;) {
                 int i = args.indexOf(EMPTY_SLOT);
                 if (i >= 0 && i+1 < args.size()
-                    && (isConstant(args.get(i+1), CONSTANT_Long) ||
-                        isConstant(args.get(i+1), CONSTANT_Double)))
+                    && (isConstant(args.get(i+1), TAG_LONG) ||
+                        isConstant(args.get(i+1), TAG_DOUBLE)))
                     args.remove(i);
                 else  break;
             }
         }
+        //New implementation using the CP API
```

Comment:
This is a similar implementation of the old methods using the ClassFile API. will be merged once fully tested and implemented